### PR TITLE
get profile data

### PR DESCRIPTION
**Pull request by MohammadAlhalaq**

### CODE REVIEW

File: client/src/components/pages/Profile/index.js [hunk 1]
```diff
@@ -1,80 +1,110 @@
...
+    try {
+      const profileData = await axios.get('/api/v1/profile/recZoOgsH458Jl04m');
+      const {
+        data: {
+          data: { fields },
+        },
+      } = profileData;
+      this.setState({ profileData: fields });
+    } catch (err) {
+      // console.log(err.statusCode); // need to handle
```
Comment (fadeomar) [inline client/src/components/pages/Profile/index.js]: use notification that I used in the filter page before to handle errors

File: client/src/components/pages/Profile/index.js [hunk 2]
```diff
@@ -1,80 +1,110 @@
...
+    const { profileData } = this.state;
+    if (profileData) {
+      const {
+        profileData: {
+          avalibility,
+          city,
+          email,
+          fullName,
+          image,
+          // is_valid: isValid,
```
Comment (fadeomar) [inline client/src/components/pages/Profile/index.js]: delete comment

File: client/src/components/common/AvailabilityTime/index.js [hunk 3]
```diff
@@ -23,13 +32,7 @@ const Available = ({ availableityTime }) => {
...
 
 Available.propTypes = {
-  availableityTime: propTypes.arrayOf(
-    propTypes.shape({
-      day: propTypes.string.isRequired,
-      from: propTypes.string.isRequired,
-      to: propTypes.string.isRequired,
-    })
-  ).isRequired,
+  availableityTime: propTypes.string.isRequired,
```
Comment (hashemt) [inline client/src/components/common/AvailabilityTime/index.js]:
```suggestion
  availableityTime: PropTypes.string.isRequired,
```

Comment (Asmaathabet) [inline client/src/components/common/AvailabilityTime/index.js]:
```suggestion
  availableityTime: PropTypes.string.isRequired,
```

File: client/src/components/common/Location/index.js [hunk 4]
```diff
@@ -18,19 +18,23 @@ const Location = ({ locationInfo: { address, postalCode, map } }) => {
...
+
 Location.propTypes = {
-  locationInfo: propTypes.shape({
-    address: propTypes.string.isRequired,
-    postalCode: propTypes.string.isRequired,
-    map: propTypes.string.isRequired,
-  }).isRequired,
+  address: propTypes.string.isRequired,
+  postCode: propTypes.string.isRequired,
+  imageURL: propTypes.string.isRequired,
```
Comment (Asmaathabet) [inline client/src/components/common/Location/index.js]: change all propTypes word to PropTypes

File: client/src/components/pages/Profile/index.js [hunk 5]
```diff
@@ -1,80 +1,143 @@
-import React from 'react';
-import propTypes from 'prop-types';
+import React, { Component } from 'react';
 import { Icon, Avatar } from 'antd';
+import propTypes from 'prop-types';
+
```
Comment (Asmaathabet) [inline client/src/components/pages/Profile/index.js]: delete this line

File: client/src/components/common/AvailabilityTime/index.js [hunk 6]
```diff
@@ -1,12 +1,21 @@
...
+  const time = JSON.parse(availableityTime);
+
+  const tableData = time
+    .filter(item => item.from && item.to)
+    .map((element, index) => ({
+      day: element.day,
+      from: element.from,
+      to: element.to,
+      key: index,
+    }));
```
Comment (leggomuhgreggo) [inline client/src/components/common/AvailabilityTime/index.js]: Beautiful. I actually feel joy when seeing well crafted array methods.

Couple Bonus Consideration Links:
* [Point-Free Style](https://www.freecodecamp.org/news/how-point-free-composition-will-make-you-a-better-functional-programmer-33dcb910303a/)
* [Transducer](https://medium.com/javascript-scene/transducers-efficient-data-processing-pipelines-in-javascript-7985330fe73d)

File: client/src/components/pages/Profile/index.js [hunk 7]
```diff
@@ -1,80 +1,150 @@
...
+  state = {
+    profileData: '',
+  };
+
+  async componentDidMount() {
+    const { history } = this.props;
+    try {
+      const {
+        match: {
+          params: { id },
```
Comment (leggomuhgreggo) [inline client/src/components/pages/Profile/index.js]: +1 for pro-level nested destructuring

File: client/src/components/pages/Profile/index.js [hunk 8]
```diff
@@ -1,80 +1,150 @@
...
+        },
+      } = this.props;
+
+      const profileData = await axios.get(`/api/v1/profile/${id}`);
```
Comment (leggomuhgreggo) [inline client/src/components/pages/Profile/index.js]: Getting API request code in a centralized abstraction helps avoid that [shotgun surgery](https://refactoring.guru/smells/shotgun-surgery) -- something along the lines of:



```js
// api/scrinc-client.js

const API_BASE_PATH = process.env.API_BASE_PATH || '';

export const getProfile = (id) => axios.get(`${API_BASE_PATH}/api/v1/profile/${id}`);


// Profile/index.js

import * as shrincClient from ''api/scrinc-client.js"

shrincClient.getProfile(id)
```

File: client/src/components/pages/Profile/index.js [hunk 9]
```diff
@@ -1,80 +1,150 @@
...
+      const {
+        data: {
+          data: { fields },
+        },
+      } = profileData;
+      this.setState({ profileData: fields });
+    } catch (err) {
+      const error500 = err.message.includes('500');
```
Comment (leggomuhgreggo) [inline client/src/components/pages/Profile/index.js]: 👍 All about the abstraction -- gotta fight them magic strings

I think [based on this](https://github.com/axios/axios#handling-errors) `error.response.status` is discrete to the HTTP code, if you felt like skipping the includes in favor off equivalency, but fairly inconsequential either way.

File: client/src/components/pages/Profile/index.js [hunk 10]
```diff
@@ -1,80 +1,150 @@
...
-        </div>
-      </section>
-      <Location locationInfo={locationInfo} />
-    </section>
-  );
-};
+        </section>
+      );
+    }
+    return 'loading';
```
Comment (leggomuhgreggo) [inline client/src/components/pages/Profile/index.js]: I recommend inverting the conditional logic to use an early return for 'loading' -- avoids nesting and makes it a little easier to understand at a glance -- ex:

```js
if (!profileData) {
  return 'loading';
}
...less nested component code...
```

File: client/src/components/pages/Profile/index.js [hunk 11]
```diff
@@ -1,80 +1,150 @@
...
+  };
+
+  render() {
+    return <div>{this.renderCode()}</div>;
```
Comment (leggomuhgreggo) [inline client/src/components/pages/Profile/index.js]: +1 for use of render functions to help segment the code.

Comment (MohammadAlhalaq) [reply client/src/components/pages/Profile/index.js]: Thank you

File: client/src/components/pages/Profile/index.js [hunk 12]
```diff
@@ -1,80 +1,148 @@
...
+        </section>
+        <section className="profile_contact_container">
+          <h2>
+            <Icon type="phone" /> Contact Info
+          </h2>
+          <div className="profile_contact_info">
+            <h4>Email</h4>
+            <p>{email}</p>
+            {skype ? (
+              <>
```
Comment (fadeomar) [inline client/src/components/pages/Profile/index.js]: use ```<fragment>```

Comment (fadeomar) [reply client/src/components/pages/Profile/index.js]: and use logic condition  better than ternary operator

File: client/src/components/pages/Profile/index.js (hunk 7 above)
Comment (MohammadAlhalaq) [reply client/src/components/pages/Profile/index.js]: Thank's

File: client/src/components/pages/Profile/index.js (hunk 9 above)
Comment (MohammadAlhalaq) [reply client/src/components/pages/Profile/index.js]: Nice.

File: client/src/components/pages/Profile/index.js (hunk 10 above)
Comment (MohammadAlhalaq) [reply client/src/components/pages/Profile/index.js]: Done

**Pull request merged**